src/components/register/RegisterCard.vue

```diff
@@ -0,0 +1,212 @@
+<template>
+  <div class="register-card-wrapper">
+    <form @submit.prevent="submitForm" novalidate>
+      <div
+        class="register-card-wrapper__form form-field"
+        :class="{ 'form-field--error': !!emailError }"
+      >
+        <MainInput v-model="email" @blurTouched="emailTouched = true" />
+        <div>
+          {{ emailError }}
+        </div>
+        <PasswordInput
+          v-model="password"
+          @blurTouched="passwordTouched = true"
+        />
+        <div>
+          {{ passwordError }}
+        </div>
+        <p
+          :class="{
+            password_error: !passwordEightCharacters && passwordTouched,
+          }"
+        >
+          At least 8 characters
+        </p>
+        <p :class="{ password_error: !passwordOneLetter && passwordTouched }">
+          At lest one letter
+        </p>
+        <p :class="{ password_error: !passwordOneDigit && passwordTouched }">
+          At least one digit
+        </p>
+        <div class="register-card-wrapper__form__buttons">
+          <MainButton button-type="secondary">Log in instead</MainButton>
+          <MainButton button-type="primary" :disabled="!isFormValid"
+            >Next step</MainButton
+          >
+        </div>
+      </div>
+    </form>
+  </div>
+</template>
+
+<script>
+import MainButton from "../UI/MainButton.vue";
+import MainInput from "../UI/EmailInput.vue";
+import PasswordInput from "../UI/PasswordInput.vue";
+
+export default {
+  data() {
+    return {
+      email: "",
+      password: "",
+      emailTouched: false,
+      passwordTouched: false,
+      passwordEightCharacters: false,
+      passwordOneLetter: false,
+      passwordOneDigit: false,
+    };
+  },
+  methods: {
+    registerUser() {
+      console.log("user email: ", this.email);
+      console.log("user password: ", this.password);
+    },
+    submitForm() {
+      this.touchAll();
+      if (this.isFormValid) {
+        console.log("submitted");
+        this.registerUser();
+      }
+    },
+    touchAll() {
+      this.emailTouched = true;
+      this.passwordTouched = true;
+    },
+    isEmailValid(email) {
+      let emailRegEx =
+        /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+      return email && emailRegEx.exec(email);
+    },
+    isValidPassword(password) {
+      console.log(password);
+      console.log(
+        this.passwordEightCharacters,
+        this.passwordOneDigit,
+        this.passwordOneLetter
+      );
+      if (this.password.length > 7) {
+        this.passwordEightCharacters = true;
+      } else {
+        this.passwordEightCharacters = false;
+      }
```

Comment:
I suggest to move each password validation to a separae computed value (instead of keeping it in the `data`) e.g.
```js
passwordEightCharacters() {
  return this.password.length > 7
}
```

`isValidPassword` may be another computed value (instead of a method)